### DISCUSSION TITLE
proot and chroot

### DISCUSSION THREAD

**Discussion by DocMAX** [termux/termux-app#3780] [General]
I try to use chroot on a proot directory. the os has user u0_a289. can chroot use this directoy wthout having to do chown root:root? isn't fakechroot what this is supposed to do? any ideas?

**Comment by sylirre** [comment]
`chroot` works independently of ownership. This utility just changes root file system directory in current context and spawns new process. But for proper chroot environments the rootfs directory must be owned (chown) by `root:root` and have access mode 755 (chmod).

If you are using `chroot` on a proot directory, don't expect everything to work correctly. As ownership and capabilities vanished, programs using setuid/setgid and Linux capabilities won't work properly when you switch to normal user.

`fakechroot` is supposed to be used by regular user, therefore directory and files inside must have current user as owner.
